csharp/src/Drivers/Apache/Hive2/HiveServer2Statement.cs

```diff
@@ -587,5 +592,370 @@ protected internal QueryResult EnhanceGetColumnsResult(Schema originalSchema, IR
 
             return new QueryResult(rowCount, new HiveServer2Connection.HiveInfoArrowStream(enhancedSchema, enhancedData));
         }
+
+        // Helper method to read all batches from a stream
+        private async Task<(List<RecordBatch> Batches, Schema Schema, int TotalRows)> ReadAllBatchesAsync(
+            IArrowArrayStream stream, CancellationToken cancellationToken)
+        {
+            List<RecordBatch> batches = new List<RecordBatch>();
+            int totalRows = 0;
+            Schema schema = stream.Schema;
+            
```

Comment:
The whitespace linter is still complaining about lines like this. If you're using the original Visual Studio, you can autoformat with Control-K Control-D. If you're using VS Code, you can autoformat with Alt-Shift-F. Either one will remove the trailing spaces. Other editors may have similar commands.